### PR TITLE
[#5145] Maxxed and restart aggregation jobs

### PR DESCRIPTION
**Pull request by zuhdil**
# TODO / Done

Summarize what has been changed / what has to be done in order to finalize the PR.

 - [x] Backend - Endpoint to restart aggregation jobs
 - [x] Backend - Limit auto-reschedule attempts and mark them as MAXXED
 - [x] Frontend - Implement restart button

# Test plan

What tests are necessary to ensure this works or doesn't break anything working

 - [x] Unit tests
 - [ ] Manual testing

### CODE REVIEW

File: akvo/rsr/tests/usecases/jobs/test_aggregation.py [hunk 1]
```diff
@@ -211,3 +212,23 @@ def test_success_after_failure(self):
         msg = mail.outbox[1]
         self.assertEqual(msg.to, [self.user.email])
         self.assertEqual(msg.subject, "Previously failed indicator aggregation job has succeeded")
+
+
+class HandleFailedJobTestCase(AggregationJobBaseTests):
+    def setUp(self):
+        super().setUp()
+        for _ in range(usecases.MAX_ATTEMPTS - 1):
+            self.job.mark_failed()
```
Comment (MichaelAkvo) [inline akvo/rsr/tests/usecases/jobs/test_aggregation.py:221]: Why not set the status and attempts directly?

**Comment by MichaelAkvo** [comment]
For @ifirmawan 

# Wording

image

image

I missed it in the last PR, but I don't think the user needs to know it's a "Cron job". Just "job" should be sufficient

image


For the history, I'd call it "Aggregation Job History". Users probably don't know what a "cron job" is.

------------

image

This reads like a prompt for an action. I'd write "Too many failed attempts".

# Visual

image

The last element in the history doesn't need a line. It falsely indicates that something else is there.

**Comment by ifirmawan** [comment]
> For @ifirmawan
> 
> # Wording
> image
> 
> image
> 
> I missed it in the last PR, but I don't think the user needs to know it's a "Cron job". Just "job" should be sufficient
> 
> image
> 
> For the history, I'd call it "Aggregation Job History". Users probably don't know what a "cron job" is.
> 
> image
> 
> This reads like a prompt for an action. I'd write "Too many failed attempts".
> 
> # Visual
> image
> 
> The last element in the history doesn't need a line. It falsely indicates that something else is there.

I just fix this issue by this commit [ff69da3](https://github.com/akvo/akvo-rsr/pull/5158/commits/ff69da32b62cf94819607988221a2ab7c74133bb)

File: akvo/rsr/spa/app/modules/program/ActualValue.jsx [hunk 2]
```diff
@@ -1,43 +1,61 @@
...
       {_status && (
         <Aggregation.Col icon>
-          <Aggregation.Tooltip title={title}>
-            {
-              (callToAction.includes(job?.status) && job?.id)
-                ? (
-                  <Button shape="circle" onClick={handleOnRestartJob}>
-                    <Icon type={iconType} width="16px" height="16px" className={job.status} />
+          {
+            (callToAction.includes(job?.status) && job?.id && job?.attempts < MAX_ATTEMPTS)
```
Comment (MichaelAkvo) [inline akvo/rsr/spa/app/modules/program/ActualValue.jsx:45]: Why not use only the job status here? :thinking:  Why also `MAX_ATTEMPTS`? @ifirmawan

Comment (ifirmawan) [reply akvo/rsr/spa/app/modules/program/ActualValue.jsx:45]: I thought that would be covered when the job nears the max attempt, but if it's not necessary then I will remove it

**Review by MichaelAkvo — APPROVED** [review]
Tested as working locally. Looks good to me

**Comment by MichaelAkvo** [comment]
Closed #5145

**Pull request merged**